app/components/map-component/index.tsx

```diff
@@ -0,0 +1,210 @@
+
+  // toggle modal from inside modal component instead of here in the parent
+  const toggleModal = React.useCallback(
+    () => openSettingsModalRef.current?.toggleVisibility(),
+    [],
+  )
+
+  const respondToBlocked = (status: PermissionStatus) => {
+    // iOS will only ever ask once for permission, and initial checks can differentiate between BLOCKED vs DENIED
+    if (isIOS) {
+      if (permissionsStatus === RESULTS.BLOCKED && status === RESULTS.BLOCKED) {
+        toggleModal()
+      }
+      // Android can ask twice for permission, and initial checks cannot differentiate between BLOCKED vs DENIED
+    } else {
+      // TODO figure out how to not fire this after the 2nd time user blocks
+      toggleModal()
+    }
+  }
+
+  const centerOnUser = () => {
+    getUserRegion(async (region) => {
+      if (region && mapViewRef.current) {
+        mapViewRef.current.animateToRegion(region)
+      } else {
+        onError()
+      }
+    })
+  }
+
+  const requestLocationPermission = async () => {
+    request(LOCATION_PERMISSION)
+      .then((status) => {
+        if (status === RESULTS.GRANTED) {
+          centerOnUser()
+        } else if (status === RESULTS.BLOCKED) {
+          respondToBlocked(status)
+        }
+        setPermissionsStatus(status)
+      })
+      .catch(() => onError())
+  }
+
+  const debouncedHandleRegionChange = React.useRef(
+    debounce(
+      (region: Region) => {
+        updateMapLastCoords(client, {
+          latitude: region.latitude,
+          longitude: region.longitude,
+        })
+      },
+      1000,
+      { trailing: true },
+    ),
+  ).current
+
+  return (
+    <View style={styles.viewContainer}>
+      <MapView
+        ref={mapViewRef}
+        style={styles.map}
+        showsUserLocation={permissionsStatus === RESULTS.GRANTED}
+        showsMyLocationButton={false}
+        initialRegion={userLocation}
+        customMapStyle={themeMode === "dark" ? MapStyles.dark : MapStyles.light}
+        onPress={handleMapPress}
+        onRegionChange={debouncedHandleRegionChange}
+        onMarkerSelect={(e) => {
+          // react-native-maps has a very annoying error on iOS
+          // When two markers are almost on top of each other onSelect will get called for a nearby Marker
+          // This improvement (not an optimal fix) checks to see if that error happened, and quickly reopens the correct callout
+          const matchingLat =
+            e.nativeEvent.coordinate.latitude ===
+            focusedMarker?.mapInfo.coordinates.latitude
+          const matchingLng =
+            e.nativeEvent.coordinate.longitude ===
+            focusedMarker?.mapInfo.coordinates.longitude
+          if (!matchingLat || !matchingLng) {
+            if (focusedMarkerRef.current) {
+              focusedMarkerRef.current.showCallout()
+            }
+          }
+        }}
+      >
+        {(data?.businessMapMarkers ?? []).reduce(
```

Comment:
this PR will help using a .map instead: https://github.com/GaloyMoney/galoy/pull/3856